bson/src/main/org/bson/codecs/configuration/CodecProvider.java

```diff
@@ -34,10 +38,34 @@ public interface CodecProvider {
     /**
      * Get a {@code Codec} using the given context, which includes, most importantly, the Class for which a {@code Codec} is required.
      *
+     * <p>This method is called only if {@link #get(Class, List, CodecRegistry)} is not properly overridden.</p>
+     *
      * @param clazz the Class for which to get a Codec
      * @param registry the registry to use for resolving dependent Codec instances
      * @param <T> the type of the class for which a Codec is required
      * @return the Codec instance, which may be null, if this source is unable to provide one for the requested Class
      */
     <T> Codec<T> get(Class<T> clazz, CodecRegistry registry);
```

Comment:
The driver no longer calls the "old" `CodecProvider.get`, i.e., we may even deprecate this method in the future and remove in 5.0, thus going back to having only one method `CodecProvider`.